### PR TITLE
Add --clean to avoid loading ftplugins

### LINKED ISSUE #105: Vroom is using settings in ftplugin/ directory
Although vroom behavior isn't affected by `.vimrc` configurations, it looks it can be affected by personal configuration lying under `~/.vim`, such as `~/.vim/ftplugin`. This is a big concern when testing things like filetype plugins, but it could potentially raise an issue for any sort of plugin (ex. I'm unintentionally depending on a script in `.vim/autoload`).

This doesn't seem to be vroom-specific, since `vim -u NONE` still includes `~/.vim` in the `rtp`. However since this has a big impact on vroom, is there any way vroom can step around this vim behavior? If not, maybe there should be some sort of warning in the README.

### PR DESCRIPTION
**Pull request by nfischer**
This uses vim's '--clean' switch to avoid loading ftplugins, since this
can affect behavior during vroom tests. This switch seems to avoid the
issue in my local testing, whereas other things ('-i NONE',
'--noplugin', etc.) don't seem to have the same effect.

I can't get vroom running in '--neovim' mode (nvim 0.4.4), so this doesn't
attempt to address the issue in the neovim code path. neovim appears to
define '--clean' differently from vim, so it's possible neovim may
require a different fix.

I verified this still respects vroom's '--vimrc' switch while having the
desired behavior of avoiding plugins.

Fixes #105

### CODE REVIEW

**Comment by nfischer** [comment]
@dbarnett This seems to work correctly for vim 8.1.2269 (the particular test case I had issues with passes, as do all the other vroom test cases I'm aware of). Can you advise on how to check availability of the '--clean' switch? Is there an easy way to check the vim version?

**Comment by dbarnett** [comment]
> Can you advise on how to check availability of the '--clean' switch? Is there an easy way to check the vim version?

I would suggest to always attempt to start with `--clean` but on errors fall back to running without `--clean`. That way in the common case there's no overhead.

Alternatives would be to always run `vim --version` first (guaranteed overhead, and could have gotchas trying to check that way anyway), or to accept new `--clean` and `--noclean` args that the caller has to configure correctly (awkward for users and still takes effort to implement).

**Comment by nfischer** [comment]
PTAL. It looks like the default wait time is 500ms which is more than enough for vim to recognize unknown switches. I tested this locally by changing `--clean` to `--someunknownarg` (to force this error case) and the fallback code worked as expected 10/10.

File: vroom/vim.py [hunk 1]
```diff
@@ -120,6 +124,13 @@ def Start(self):
     # still has a _vim attribute it can query for details.
     self.process = subprocess.Popen(self.start_command, env=self.env)
     time.sleep(self.args.startuptime)
+    if self.process.poll() is not None and self.process.poll() != 0:
+      # If vim exited this quickly, it probably means we passed a switch it
+      # doesn't recognize. Try again without the '--clean' switch since this is
+      # new in 8.0.1554+.
+      self.start_command.remove('--clean')
```
Comment (dbarnett) [inline vroom/vim.py:131]: FYI, I'm thinking of eventually plumbing through and post a warning notice in the final output if we couldn't use `--clean`.

**Comment by dbarnett** [comment]
That'll do the trick! Thanks!

Merging.

**Pull request merged**